### PR TITLE
214106: Azure Front Door Update

### PR DESCRIPTION
**Pull request by osagiestar**
# Description
Created bicep template and associated parameter file
Added the template for Azure Front Door deployment in the deploy-platform-core.yaml file
Updated Front Door registry module
Removed the 'Standard_AzureFrontDoor' option from sku

Link to the relevant work items: e.g. [AB#214106](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/214106)
[[AB#214807](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/214807)] 

# Testing
Testing as already setup within the ADO pipeline runs.

### CODE REVIEW

File: infra/front-door/front-door.bicep [hunk 1]
```diff
@@ -0,0 +1,63 @@
+@description('Required. Name of Front Door Profile.')
+param name string
+
+@description('Required. The pricing tier of the FrontDoor profile.')
+@allowed([
+  'Standard_AzureFrontDoor'
```
Comment (DaRoszko) [inline infra/front-door/front-door.bicep:6]: remove standard

Comment (osagiestar) [reply infra/front-door/front-door.bicep]: Sorted

File: infra/front-door/front-door.parameters.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "name": {
+      "value": "#{{ infraResourceNamePrefix }}#{{ nc_resource_frontdoor }}#{{ nc_instance_regionid }}01"
+    },
+    "sku": {
+      "value": "Premium_AzureFrontDoor"
```
Comment (ajaysoni-cap) [inline infra/front-door/front-door.parameters.json:9]: This should use the environment-specific variable

File: infra/front-door/front-door.bicep [hunk 3]
```diff
@@ -0,0 +1,44 @@
+@description('Required. Name of Front Door Profile.')
+param name string
+
+@description('Required. The pricing tier of the FrontDoor profile.')
+@allowed([
+  'Premium_AzureFrontDoor'
+])
```
Comment (ajaysoni-cap) [inline infra/front-door/front-door.bicep:7]: Maybe should have all the available SKUs / we should not have allowed attribute
The reason is, that if we control the SKU from environment-specific variables, just changing the variable will not have the template working.

File: .azuredevops/deploy-platform-core.yaml [hunk 4]
```diff
@@ -99,6 +99,11 @@ extends:
             privateEndpointDnsRecordsForResources:
               - resourceName: $(infraResourceNamePrefix)$(nc_resource_servicebus)$(nc_instance_regionid)01
                 resourceGroupName: $(servicesResourceGroup)
+      - name: front_door
```
Comment (ajaysoni-cap) [inline .azuredevops/deploy-platform-core.yaml:102]: Do we really need a separate job for just the front door?
Should this be added to the network job? @ppotturi @DaRoszko

**Review by DaRoszko — APPROVED** [review]

**Review by DaRoszko — APPROVED** [review]

File: .azuredevops/deploy-platform-core.yaml (hunk 4 above)
Comment (ajaysoni-cap) [reply .azuredevops/deploy-platform-core.yaml:102]: As discussed, we will revisit this later when other components are deployed from the same pipeline.

File: infra/front-door/front-door.bicep (hunk 3 above)
Comment (ajaysoni-cap) [reply infra/front-door/front-door.bicep:7]: As discussed, we agreed that the SKU for the front door is not likely to change for any environment.

**Review by ajaysoni-cap — APPROVED** [review]

**Review by ganeshg2491 — APPROVED** [review]

**Pull request merged**